sdk/core/AzureCore/Source/Pipeline/Policies/ContentDecodePolicy.swift

```diff
@@ -102,17 +102,19 @@ internal class ContentDecodeXMLParser: NSObject, XMLParserDelegate {
             for (key, value) in attributeDict {
                 let defaultKey = "_\(key)"
                 let jsonKey = xmlMap?[defaultKey]?.jsonName ?? defaultKey
-                let attr = XMLTreeNode(name: jsonKey, type: .property, parent: newNode, value: value)
+                let attr = XMLTreeNode(name: jsonKey, type: .property, parent: nil, value: value)
```

Comment:
This was the source of the XMLTree memory leak.

Reply:
Is it worth adding a code comment about the fact that parent is set to nil explicitly to avoid a memory leak (and maybe explain why not doing so causes the leak)?

Reply:
TL/DR: I don't think a comment would help here--the assert is the best defense.

There are many places where not handling the parent reference correctly could introduce a memory leak, not just here, which is why I put the assert statement in XMLModel. The test cases and the demo application will fail if a regression is introduced and from that point it should be easy to diagnose and correct the problem.

In fact, that's exactly what happened. As soon as I introduced the assert I saw that the underscored properties (which come from XML attributes) were holding onto a parent reference and creating a memory leak.

Reply:
Excellent explanation, thanks! And now it's at least archived on this PR for the future :)